### PR TITLE
Fix request method for answer

### PR DESCRIPTION
**Pull request by toothfairy**
When answer to quiz questions, request for answer should be quiz

### CODE REVIEW

**Comment by jgorset** [comment]
Is the quiz really the thing that it's an answer to, though? I'd argue that it's an answer to the question, and the question just happens to be part of a quiz.

**Comment by jgorset** [comment]
Why do you need the `request` method for an answer to a question that is part a quiz to return the quiz instead of the question (like an answer to a question that isn't part of a quiz would do)?

File: app/models/tasuku/taskables/question/answer.rb [hunk 1]
```diff
@@ -15,7 +15,13 @@ def question
         votes.first.option.question if votes.any?
       end
 
-      request is: :question
+      def request
+        if question.quizzes.present?
+          question.quizzes.first
```
Comment (jgorset) [inline app/models/tasuku/taskables/question/answer.rb:20]: This smells a bit.

**Comment by toothfairy** [comment]
> Why do you need the request method for an answer to a question that is part a quiz to return the quiz instead of the question (like an answer to a question that isn't part of a quiz would do)?

The idea is that when we subscribe to observe submission, submission request returns us quiz (not questions). And we don't need to make extra checks if question belongs_to quiz and this makes observer logic more consistent.

File: app/models/tasuku/taskables/question/answer.rb (hunk 1 above)
Comment (toothfairy) [inline app/models/tasuku/taskables/question/answer.rb:20]: Current design is that question can belong to multiple quizes and be stand alone question at the same time. But such scenario breaks submission process and proper statistics calculation. This has to be reworked somewhen, but the fix is work around for now

**Comment by jgorset** [comment]
Why do you need it to return the quiz instead of the question, though? What does your observer do?

**Comment by jgorset** [comment]
Maybe I'm just being difficult, but it seems like this makes more sense to do somewhere else. Something about `Answer.request` returning different things based on whether its question is part of a quiz just rubs me the wrong way unless there's a really good reason for why you would always want that behaviour.

**Comment by toothfairy** [comment]
So just to make it clear: answer is submittable for question, and quiz consists of multiple questions. So technically when you submit answer, answer may be submittable for either question or quiz through question. When you subscribe to answer (as a submittable) you probably want to get a request for it and check if that request is complete, right? The easiest way (and perhaps the most consistent) is just to call submittable.request. It's hard to imaging situation when you need to get question as a request for answer in context of quiz.

**Pull request closed without merging**